coordinator/internal/userapi/userapi.go

```diff
@@ -0,0 +1,279 @@
+// Copyright 2024 Edgeless Systems GmbH
+// SPDX-License-Identifier: AGPL-3.0-only
+
+// Package userapi implements the userapi.UserAPI gRPC service.
+//
+// It is responsible for
+//   - translating between RPCs and library calls
+//   - authorizing users
+//   - managing content-addressed history elements (but not the LatestTransition)
+//   - creating secrets (seed and mesh CA key)
+//
+// The package has access to an authority.Authority, which is the source of truth for the current
+// state of the system, and a history.History for storing supporting content (manifests and
+// policies).
+package userapi
+
+import (
+	"bytes"
+	"context"
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log/slog"
+	"slices"
+
+	"github.com/edgelesssys/contrast/coordinator/internal/authority"
+	"github.com/edgelesssys/contrast/internal/constants"
+	"github.com/edgelesssys/contrast/internal/crypto"
+	"github.com/edgelesssys/contrast/internal/manifest"
+	"github.com/edgelesssys/contrast/internal/seedengine"
+	"github.com/edgelesssys/contrast/internal/userapi"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
+)
+
+type stateUpdateAuthorizer = func(*manifest.Manifest) (*seedengine.SeedEngine, *ecdsa.PrivateKey, error)
+
+// auth is the public API of authority.Authority.
+type auth interface {
+	// GetState returns the current state. If the error is nil, the state must be set.
+	GetState() (*authority.State, error)
+	// GetHistory returns a slice of manifests and a map of policies referenced in the manifests.
+	GetHistory() ([][]byte, map[manifest.HexString][]byte, error)
+	// UpdateState advances the state to the given manifest and policies.
+	UpdateState(*authority.State, *seedengine.SeedEngine, []byte, [][]byte) (*authority.State, error)
```

Comment:
While parameter names can be left out, I think it is really helpful to add them in the interface definition, too.

Reply:
Named parameters